frontends/krita/krita_diff/script.py

```diff
@@ -357,11 +349,37 @@ def apply_img2img(self, is_inpaint):
             save_img(sel_image, path)
 
         def cb(response):
+            def cb_upscale(upscale_response):
+                if len(self.client.long_reqs) == 1:  # last request
+                    self.eta_timer.stop()
+                assert response is not None, "Backend Error, check terminal"
+
+                outputs = upscale_response["images"]
+                layer_name_prefix = "inpaint" if is_inpaint else "img2img"
+                glayer_name, layer_names = get_desc_from_resp(response, layer_name_prefix)
+                layers = [
+                    insert(name if name else f"{layer_name_prefix} {i + 1}", output)
+                    for output, name, i in zip(outputs, layer_names, itertools.count())
+                ]
+                if self.cfg("hide_layers", bool):
+                    for layer in layers[:-1]:
+                        layer.setVisible(False)
+                if glayer:
+                    glayer.setName(glayer_name)
+                self.doc.refreshProjection()
+                self.controlnet_transparency_mask_inserter(layers, mask_image)
```

Comment:
Check if it's inpaint before working with the transparency layer (got an error on my end).
```suggestion
                if is_inpaint:
                    self.controlnet_transparency_mask_inserter(layers, mask_image)
```